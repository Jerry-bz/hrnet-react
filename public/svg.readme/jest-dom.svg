<svg xmlns="http://www.w3.org/2000/svg" width="103.61" height="35" viewBox="0 0 103.61 35"><rect class="svg__rect" x="0" y="0" width="103.61" height="35" fill="#2F6209"/><rect class="svg__rect" x="103.61" y="0" width="0" height="35" fill="#8FC965"/><path class="svg__text" d="M13.61 19.57L13.61 19.57L15.09 19.57Q15.09 20.25 15.43 20.59Q15.76 20.93 16.40 20.93L16.40 20.93Q17.00 20.93 17.35 20.54Q17.71 20.14 17.71 19.45L17.71 19.45L17.71 13.47L19.19 13.47L19.19 19.45Q19.19 20.68 18.42 21.40Q17.66 22.12 16.40 22.12L16.40 22.12Q15.07 22.12 14.34 21.44Q13.61 20.77 13.61 19.57ZM29.45 22L23.87 22L23.87 13.47L29.40 13.47L29.40 14.66L25.35 14.66L25.35 17.02L28.85 17.02L28.85 18.19L25.35 18.19L25.35 20.82L29.45 20.82L29.45 22ZM33.21 19.42L33.21 19.42L34.69 19.42Q34.69 20.15 35.17 20.55Q35.65 20.95 36.55 20.95L36.55 20.95Q37.32 20.95 37.71 20.63Q38.10 20.32 38.10 19.80L38.10 19.80Q38.10 19.24 37.70 18.94Q37.31 18.63 36.28 18.32Q35.25 18.01 34.64 17.63L34.64 17.63Q33.47 16.90 33.47 15.72L33.47 15.72Q33.47 14.69 34.31 14.02Q35.15 13.35 36.49 13.35L36.49 13.35Q37.39 13.35 38.08 13.68Q38.78 14.01 39.18 14.61Q39.58 15.22 39.58 15.96L39.58 15.96L38.10 15.96Q38.10 15.29 37.68 14.91Q37.26 14.54 36.48 14.54L36.48 14.54Q35.76 14.54 35.35 14.85Q34.95 15.16 34.95 15.71L34.95 15.71Q34.95 16.18 35.39 16.50Q35.82 16.81 36.82 17.10Q37.81 17.40 38.42 17.78Q39.02 18.16 39.30 18.65Q39.58 19.13 39.58 19.79L39.58 19.79Q39.58 20.86 38.77 21.49Q37.95 22.12 36.55 22.12L36.55 22.12Q35.62 22.12 34.85 21.77Q34.07 21.43 33.64 20.83Q33.21 20.22 33.21 19.42ZM45.55 14.66L42.92 14.66L42.92 13.47L49.68 13.47L49.68 14.66L47.02 14.66L47.02 22L45.55 22L45.55 14.66ZM56.06 18.95L52.99 18.95L52.99 17.80L56.06 17.80L56.06 18.95ZM62.82 22L60.37 22L60.37 13.47L62.89 13.47Q64.02 13.47 64.89 13.97Q65.77 14.48 66.25 15.40Q66.73 16.33 66.73 17.52L66.73 17.52L66.73 17.95Q66.73 19.16 66.25 20.08Q65.76 21.00 64.87 21.50Q63.98 22 62.82 22L62.82 22ZM61.85 14.66L61.85 20.82L62.81 20.82Q63.98 20.82 64.60 20.09Q65.23 19.36 65.24 17.99L65.24 17.99L65.24 17.52Q65.24 16.13 64.64 15.40Q64.03 14.66 62.89 14.66L62.89 14.66L61.85 14.66ZM70.92 18.00L70.92 18.00L70.92 17.52Q70.92 16.28 71.37 15.32Q71.81 14.37 72.61 13.86Q73.42 13.35 74.46 13.35Q75.51 13.35 76.31 13.85Q77.12 14.35 77.56 15.29Q78.00 16.23 78.00 17.48L78.00 17.48L78.00 17.96Q78.00 19.21 77.57 20.16Q77.14 21.10 76.33 21.61Q75.52 22.12 74.47 22.12L74.47 22.12Q73.44 22.12 72.63 21.61Q71.81 21.10 71.37 20.17Q70.93 19.23 70.92 18.00ZM72.41 17.46L72.41 17.96Q72.41 19.36 72.95 20.13Q73.50 20.90 74.47 20.90L74.47 20.90Q75.46 20.90 75.99 20.15Q76.52 19.40 76.52 17.96L76.52 17.96L76.52 17.51Q76.52 16.09 75.98 15.34Q75.45 14.58 74.46 14.58L74.46 14.58Q73.50 14.58 72.96 15.33Q72.42 16.09 72.41 17.46L72.41 17.46ZM83.94 22L82.47 22L82.47 13.47L84.39 13.47L86.85 20.01L89.31 13.47L91.22 13.47L91.22 22L89.74 22L89.74 19.19L89.89 15.43L87.37 22L86.31 22L83.80 15.43L83.94 19.19L83.94 22Z" fill="#FFFFFF"/><path class="svg__text" d="" fill="#FFFFFF" x="116.61"/></svg>