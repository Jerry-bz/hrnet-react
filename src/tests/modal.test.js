import React from "react";
import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Modal from "react-modal-hrnet";

// Simulated function
const handleClose = jest.fn();

// Create a fake store with a reducer that returns an empty array
const store = configureStore({
  reducer: {
    employees: () => [],
  },
});

// Test the Modal component
describe("Checks the display of the Modal component", () => {
  render(
    <Provider store={store}>
      <Modal closeModal={handleClose} />
    </Provider>
  );

  // Test that the modal renders a close button
  it("should display the close button", () => {
    // Get the close button
    const closeButton = screen.getByAltText(/close-modal/i);

    // Simulate a click on the close button
    fireEvent.click(closeButton);

    // Check that the close button is in the document
    expect(closeButton).toBeInTheDocument();

    // Check that the function has been called
    expect(handleClose).toHaveBeenCalledTimes(1);

    expect(Modal).toMatchSnapshot();
  });
});
