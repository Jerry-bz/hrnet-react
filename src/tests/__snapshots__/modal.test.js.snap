// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks the display of the Modal component should display the close button 1`] = `[Function]`;
